scripts/update-message-index.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+
+import argparse
+from itertools import chain
+from pathlib import Path
+from string import Template
+from typing import NamedTuple, Sequence
+import yaml
+import jinja2
+
+
+class Version(NamedTuple):
+    major: int
+    minor: int
+    patch: int = 0
+
+    def __str__(self):
+        return ".".join(map(str, self))
+
+
+class Example(NamedTuple):
+    name: str
+    content: str
+
+
+class Entry(NamedTuple):
+    code: int
+    id: str
+    type: str
+    description: str
+    details: str
+    added: Version
+    removed: Version | None
```

Comment:
Can you double check which version is required for this type annotation please? I've always used `Optional[Version]` because *I think* this approach didn't work back then.

Reply:
Python [3.10](https://docs.python.org/3.10/whatsnew/3.10.html)

> Python 3.10 was released on October 4, 2021.

I guess that is old enough? What are the requirements for Python version in xkbcommon?

Anyway it is just a developer tool and it is not used in tests.

Reply:
Since the file uses `from __future__ import annotations`, the Python version doesn't matter for this.